sql/core/src/main/scala/org/apache/spark/sql/api/r/SQLUtils.scala

```diff
@@ -247,7 +247,7 @@ private[sql] object SQLUtils extends Logging {
     dataType match {
       case 's' =>
         // Read StructType for DataFrame
-        val fields = SerDe.readList(dis).asInstanceOf[Array[Object]]
+        val fields = SerDe.readList(dis, jvmObjectTracker = null).asInstanceOf[Array[Object]]
```

Comment:
That would be ideal but requires a major refactoring, because we use `SerDe` as a singleton in many places. And we haven't fully discussed how to handle multiple active `RBackend`s yet.